Continue tracking test-tree/misc

nfebe opened the pull request:
Removed this directory from tracking by error since most other folders in test-tree don't require tracking.

Reinstating.

mithuna commented:
I have been working on this since yesterday, but it was throwing warnings as we are using special characters. 
`PHP Warning:  copy(/home/mithuna/Personal Projects/permanent-rclone-qa/generatedFiles\new/file.txt): Failed to open stream: No such file or directory in /home/mithuna/Personal Projects/permanent-rclone-qa/testTree.php on line 26
`
In windows, it was complaining about all most all the characters. I tried using back slash, htmlspecialchars and base 64 encoding, But none eliminated the warning completely. I switched over to the Linux machine and it is throwing error only for 2 files. Let me know if these warnings can be removed as well.

kfogel commented on testTree.php:
```diff
@@ -1,13 +1,36 @@
-   $addStartCharacter = $inputCharacter.$filenameFirst.$filenameSecond;
-   $addEndCharacter = $filenameFirst.$filenameSecond.$inputCharacter;
-   array_push($outputFilename,$addMiddleCharacter, $addStartCharacter, $addEndCharacter);
+    $addMiddleCharacter = $filenameFirst.htmlspecialchars($inputCharacter).$filenameSecond;
```
Why is `htmlspecialchars()` being called?  My understanding is that it encodes a string as an HTML entity, but HTML is irrelevant to us here.  We're interacting with the operating system and filesystem, not with an HTML web page.  We might need to quote some characters in order to have PHP's `copy()` function create a destination file with the desired name (and `/` probably can't be used in a filename at all on most systems, but that's okay, we'll get every other character).  I'm not sure what the way to do that is in PHP, but I'm pretty sure `htmlspecialchars()` can't be it :-).

kfogel replied:
I just ran the script -- it does indeed generate files whose names contain HTML entities instead of the actual characters those entities represent:

```
'\$newfile.txt'
'&#039;newfile.txt'
123newfile.txt
'&amp;newfile.txt'
'&gt;newfile.txt'
'&lt;newfile.txt'
'new\$file.txt'
'new&#039;file.txt'
new123file.txt
'new&amp;file.txt'
'newfile\$.txt'
'newfile&#039;.txt'
newfile123.txt
'newfile&amp;.txt'
'newfile&gt;.txt'
'newfile&lt;.txt'
' newfile.txt'
'!newfile.txt'
'#newfile.txt'
%newfile.txt
'*newfile.txt'
+newfile.txt
-newfile.txt
:newfile.txt
'=newfile.txt'
'?newfile.txt'
'@newfile.txt'
'\newfile.txt'
'`newfile.txt'
{newfile.txt
'|newfile.txt'
}newfile.txt
'new file.txt'
'new!file.txt'
new#file.txt
new%file.txt
'new*file.txt'
new+file.txt
new-file.txt
new:file.txt
'new=file.txt'
'new?file.txt'
'new@file.txt'
'new\file.txt'
'new`file.txt'
new{file.txt
'new|file.txt'
new}file.txt
'newfile .txt'
'newfile!.txt'
newfile#.txt
newfile%.txt
'newfile*.txt'
newfile+.txt
newfile-.txt
newfile.txt
newfile:.txt
'newfile=.txt'
'newfile?.txt'
'newfile@.txt'
'newfile\.txt'
'newfile`.txt'
newfile{.txt
'newfile|.txt'
newfile}.txt
'new&gt;file.txt'
'new&lt;file.txt'
```

kfogel commented on testTree.php:
```diff
@@ -1,13 +1,36 @@
+    catch(Exception $e){
+        throw new Exception('File cannot be copied! ',$e);
+    }
+    array_push($outputFilename,$addMiddleCharacter, $addStartCharacter, $addEndCharacter);
```
What is this `array_push()` for, given that we've already created the files?

kfogel commented on .gitignore:
```diff
@@ -0,0 +1 @@
+/generatedFiles/
```
Does this work with the "/" on the front and back?  I was surprised by that... what happens when you run `git status` in the directory with the `generatedFiles` subdir existing?

kfogel replied:
Ah, I just tested it -- `/generatedFiles/` does work.  I guess the front slash means "top of this git tree" and the trailing slash means "ignore this item if it is a folder" (because if you replace it with a file of the same name, then it is *not* ignored, interestingly).

kfogel commented:
> I tried using back slash, htmlspecialchars and base 64 encoding to avoid the warnings, But nothing eliminated the warning completely. I switched over to the Linux machine and it is throwing warnings only for 2 files. Let me know if these warnings can be removed as well.

The point is not just to eliminate the warnings.  It's to make sure that the right files get created on disk too!

mithuna commented:
Closing this PR as irrelevant

nfebe committed:
Continue tracking test-tree/misc

Removed this directory from tracking by error since most
 other folders in test-tree don't require tracking.

Reinstating.

Signed-off-by: Fon E. Noel NFEBE <fenn25.fn@gmail.com>
